packages/calcite-components/src/components/menu-item/menu-item.tsx

```diff
@@ -269,6 +269,11 @@ export class CalciteMenuItem implements LoadableComponent, T9nComponent, Localiz
     const { hasSubmenu, href, layout, open, submenuItems } = this;
     const key = event.key;
     const targetIsDropdown = event.target === this.dropdownActionEl;
+
+    if (event.defaultPrevented) {
+      return;
+    }
+
     if (key === " " || key === "Enter") {
       if (hasSubmenu && (!href || (href && targetIsDropdown))) {
         this.open = !open;
```

Comment:
In this example, shouldn't we be preventing default for enter and space key regardless?

Currently, it is only preventing default if the key is space, component has href and target is a dropdown.

```ts
if (key === " " || key === "Enter") {
      if (hasSubmenu && (!href || (href && targetIsDropdown))) {
        this.open = !open;
      }
      if (!(href && targetIsDropdown) && key !== "Enter") {
        this.selectMenuItem(event);
      }
      if (key === " " || (href && targetIsDropdown)) {
        event.preventDefault();
      }
    } 
```

Reply:
There are if/else statement blocks that won't be executed (e.g., `Enter` + `!hasSubmenu`), so I wouldn't agree we should cancel regardless of key pressed.